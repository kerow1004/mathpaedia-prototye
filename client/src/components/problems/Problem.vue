<template>
  <b-list-group-item>
    <div v-if="content" class="problem">
      <part :part="content" :index="id"></part>

      <b-list-group v-if="problem.parts">
        <b-list-group-item v-for="(part, partIndex) in problem.parts"
                           :key="`${problem._id}-${partIndex}`">
          <part :part="part" :index="partIndex"></part>
        </b-list-group-item>
      </b-list-group>
    </div>
  </b-list-group-item>
</template>

<script>
import Part from './Part';

export default {
  name: 'Problem',
  props: ['problem'],
  components: {
    part: Part,
  },
  computed: {
    id() {
      /* eslint-disable */
      return this.problem._id;
      /* eslint-enable */
    },
    content() {
      return this.problem.content;
    },
  },
};
</script>

<style scoped>

</style>
