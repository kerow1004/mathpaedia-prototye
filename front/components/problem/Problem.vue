<template>
  <v-card xs12>
    <div v-if="content" class="problem">
      <part :part="content" :index="id"></part>

      <v-card v-if="problem.parts && problem.parts.length > 0">
        <v-container>
          <v-flex xs12>
            <div v-for="(part, partIndex) in problem.parts"
                               :key="`${problem._id}-${partIndex}`">
                <part :part="part" :index="partIndex"></part>
            </div>
          </v-flex>
        </v-container>
      </v-card>
    </div>
  </v-card>
</template>

<script>
import Part from '@/components/problem/Part';

export default {
  name: 'Problem',
  props: ['problem'],
  components: {
    part: Part,
  },
  computed: {
    id() {
      if (!this.problem) {
        return;
      }
      /* eslint-disable */
      return this.problem._id;
      /* eslint-enable */
    },
    content() {
      /* eslint-disable */
      if (!this.problem) {
        return;
      } else {
        return this.problem.content;
      }
      /* eslint-enable */
    },
  },
};
</script>

<style scoped>

</style>
