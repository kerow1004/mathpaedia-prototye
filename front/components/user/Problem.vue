<template>
  <v-container id="problemContainer">
    <v-layout align-center justify-center row wrap>
      <v-flex class="title" xs12 sm12 md12>
        <h3>Problems</h3>
      </v-flex>
      <problem-lists-component :lists="questions"></problem-lists-component>
      <div class="centerLine"></div>
      <problem-lists-component :lists="answers"></problem-lists-component>
    </v-layout>
  </v-container>
</template>

<script>
import ProblemListsComponent from '@/components/user/ProblemLists';

export default {
  name: 'Problem',
  props: ['questions', 'answers'],
  components: {
    'problem-lists-component': ProblemListsComponent,
  },
};
</script>

<style scoped lang="scss">
$container-border-color: #bfbfbf;
$container-bg-color: #ffffff;

#problemContainer {
  background-color: $container-bg-color;
  border: 1px solid $container-border-color;
  border-top: 3px solid $container-border-color;
  padding: 0;
  .title {
    padding: 10px;
    border-bottom: 1px solid $container-border-color;
    h3 {
      font-size: 1em;
      font-weight: 400;
    }
  }
  .centerLine {
    border-top: 1px dashed $container-border-color;
    height: 1px;
    width: 90%;
  }
}
</style>
