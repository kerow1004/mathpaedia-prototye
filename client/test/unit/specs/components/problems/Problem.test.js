import { mount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Problem from '@/components/problems/Problem';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Math Problem.vue', () => {
  const id = 10;
  const content = {
    sentence: 'math problem sentence',
    picture: 'http://picture.math.com/123456',
    choices: ['1', '2', '3', '4', '5'],
  };
  const parts = [
    {
      sentence: 'math part sentence',
      picture: 'http://picture.math.com/1234567',
      choices: ['11', '22', '33', '44', '55'],
    },
  ];

  let wrapper;

  beforeEach(() => {
    wrapper = mount(Problem, {
      propsData: {
        problem: {
          _id: id,
          content,
          parts,
        },
      },
      localVue,
    });
  });

  test('equals problem content to the received content', () => {
    expect(wrapper.props().problem.content).toBe(content);
  });

  test('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
