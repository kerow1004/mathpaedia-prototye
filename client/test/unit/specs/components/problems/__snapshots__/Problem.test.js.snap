// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math Problem.vue has the expected html structure 1`] = `
<div
  class="list-group-item"
>
  <div
    class="problem"
  >
    <div>
      <div
        class="part-sentence"
      >
        <strong>
          11
        </strong>
        . math problem sentence
  
      </div>
       
      <div
        class="picture-container"
      >
        <img
          class="picture"
          src="http://picture.math.com/123456"
        />
      </div>
       
      <div>
        <ol>
          <li>
            1
          </li>
          <li>
            2
          </li>
          <li>
            3
          </li>
          <li>
            4
          </li>
          <li>
            5
          </li>
        </ol>
      </div>
    </div>
     
    <div
      class="list-group"
    >
      <div
        class="list-group-item"
      >
        <div>
          <div
            class="part-sentence"
          >
            <strong>
              1
            </strong>
            . math part sentence
  
          </div>
           
          <div
            class="picture-container"
          >
            <img
              class="picture"
              src="http://picture.math.com/1234567"
            />
          </div>
           
          <div>
            <ol>
              <li>
                11
              </li>
              <li>
                22
              </li>
              <li>
                33
              </li>
              <li>
                44
              </li>
              <li>
                55
              </li>
            </ol>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
